sickbeard/search.py

```diff
@@ -487,8 +487,14 @@ def searchProviders(show, episodes, manualSearch=False, downCurQuality=False, ma
 
     origThreadName = threading.currentThread().name
 
-    providers = [x for x in sickbeard.providers.sortedProviderList(sickbeard.RANDOMIZE_PROVIDERS)
-                 if x.is_active() and x.enable_backlog]
+    if manual_snatch:
+        logger.log("Using manual search providers")
+        providers = [x for x in sickbeard.providers.sortedProviderList(sickbeard.RANDOMIZE_PROVIDERS)
+                    if x.is_active() and x.enable_manualsearch]
+    else:
+        providers = [x for x in sickbeard.providers.sortedProviderList(sickbeard.RANDOMIZE_PROVIDERS)
+                    if x.is_active() and x.enable_backlog]
+
```

Comment:
This section can be simplified and be made more useful with a simple change.
Add a method `.enabled_for(search_type)`
Then you can use the original code nearly unchanged:
```python
providers = [x for x in sickbeard.providers.sortedProviderList(sickbeard.RANDOMIZE_PROVIDERS)
                    if x.is_active() and x.enabled_for(cur_search_type)]
```
This has the advantage of always using the correct providers for a search type, regardless of how many search types you add.